pandas/core/dtypes/common.py

```diff
@@ -82,46 +82,243 @@ def _ensure_categorical(arr):
         except ImportError:
             _is_scipy_sparse = lambda _: False
-    return _is_scipy_sparse(array)
 
+    return _is_scipy_sparse(arr)
 
-def is_categorical(array):
-    """ return if we are a categorical possibility """
-    return isinstance(array, ABCCategorical) or is_categorical_dtype(array)
 
+def is_categorical(arr):
+    """
+    Check whether an array-like is a Categorical instance.
 
-def is_datetimetz(array):
-    """ return if we are a datetime with tz array """
-    return ((isinstance(array, ABCDatetimeIndex) and
-             getattr(array, 'tz', None) is not None) or
-            is_datetime64tz_dtype(array))
+    Parameters
+    ----------
+    arr : array-like
+        The array-like to check.
+
+    Returns
+    -------
+    boolean : Whether or not the array-like is of a Categorical instance.
+
+    Examples
+    --------
+    >>> is_categorical([1, 2, 3])
+    False
+
+    Categoricals and Series Categoricals will return True.
+
+    >>> cat = pd.Categorical([1, 2, 3])
+    >>> is_categorical(cat)
+    True
+    >>> is_categorical(pd.Series(cat))
+    True
+    """
+
+    return isinstance(arr, ABCCategorical) or is_categorical_dtype(arr)
+
+
+def is_datetimetz(arr):
+    """
+    Check whether an array-like is a datetime array-like with a timezone
+    component in its dtype.
+
+    Parameters
+    ----------
+    arr : array-like
+        The array-like to check.
 
+    Returns
+    -------
+    boolean : Whether or not the array-like is a datetime array-like with
+              a timezone component in its dtype.
+
+    Examples
+    --------
+    >>> is_datetimetz([1, 2, 3])
+    False
+
+    Although the following examples are both DatetimeIndex objects,
+    the first one returns False because it has no timezone component
+    unlike the second one, which returns True.
+
+    >>> is_datetimetz(pd.DatetimeIndex([1, 2, 3]))
+    False
+    >>> is_datetimetz(pd.DatetimeIndex([1, 2, 3], tz="US/Eastern"))
+    True
+
+    The object need not be a DatetimeIndex object. It just needs to have
+    a dtype which has a timezone component.
+
+    >>> dtype = DatetimeTZDtype("ns", tz="US/Eastern")
+    >>> s = pd.Series([], dtype=dtype)
+    >>> is_datetimetz(s)
+    True
+    """
+
+    # TODO: do we need this function?
+    # It seems like a repeat of is_datetime64tz_dtype.
+
+    return ((isinstance(arr, ABCDatetimeIndex) and
+             getattr(arr, 'tz', None) is not None) or
+            is_datetime64tz_dtype(arr))
+
+
+def is_period(arr):
+    """
+    Check whether an array-like is a periodical index.
+
+    Parameters
+    ----------
+    arr : array-like
+        The array-like to check.
+
+    Returns
+    -------
+    boolean : Whether or not the array-like is a periodical index.
+
+    Examples
+    --------
+    >>> is_period([1, 2, 3])
+    False
+    >>> is_period(pd.Index([1, 2, 3]))
+    False
+    >>> is_period(pd.PeriodIndex(["2017-01-01"], freq="D"))
+    True
+    """
 
-def is_period(array):
```

Comment:
this was around before the ``is_datetime64tz_dtype``, so certainly could try to eliminate things (separate pr). the distinctions was meant to be that this would do inference on a list-like as well (where the _dtype would only work on an array).